common/views/themes/base/layout.ts

```diff
@@ -16,15 +16,23 @@ export const layout = `
 }
 
 @supports selector(:has(a)) {
-  body:has(dialog[open]) {
+  body:has(dialog[open]),
+  body:has([data-lock-scroll="true"]) {
     margin: 0;
     height: 100vh;
     overflow: hidden;
   }
+
+  html:has([data-scroll-smooth="true"]) {
+    @media (prefers-reduced-motion: no-preference) {
+      scroll-behavior: smooth;
```

Comment:
Doing the smooth scrolling with css rather than JS is cleaner, although it does now highlight that the page scrolls when you open an image modal – I started trying to fix that in this PR, but I think I'd rather try to work out the root cause and solve it properly